### DISCUSSION TITLE
Usage of FastURLFilter via ES JSONURLFilterWrapper

### DISCUSSION THREAD

**Discussion by sebastian-nagel** [DigitalPebble/storm-crawler#1012] [Q&A] [answered]
The [FastURLFilter](https://javadoc.io/doc/com.digitalpebble.stormcrawler/storm-crawler-core/latest/com/digitalpebble/stormcrawler/filtering/regex/FastURLFilter.html) configuration file contains a JSON array, not a JSON object. How can it be put into ES?

When I try to put a FastURLFilter configuration file into ES (as described in [JSONURLFilterWrapper](https://javadoc.io/doc/com.digitalpebble.stormcrawler/storm-crawler-elasticsearch/latest/com/digitalpebble/stormcrawler/elasticsearch/filtering/JSONURLFilterWrapper.html), ES responds with an error ([indicating that ES expects a JSON object](https://stackoverflow.com/questions/62778983/compressor-detection-can-only-be-called-on-some-xcontent-bytes-or-compressed-xc)):
```
curl -XPUT 'localhost:9200/config/config/fast.urlfilter.json?pretty' -H 'Content-Type: application/json' -d @fast.urlfilter.json
{
  "error" : {
    "root_cause" : [
      {
        "type" : "mapper_parsing_exception",
        "reason" : "failed to parse"
      }
    ],
    "type" : "mapper_parsing_exception",
    "reason" : "failed to parse",
    "caused_by" : {
      "type" : "not_x_content_exception",
      "reason" : "Compressor detection can only be called on some xcontent bytes or compressed xcontent bytes"
    }
  },
  "status" : 400
}
```

**Comment by jnioche** [comment, marked as answer]
Thanks @sebastian-nagel, I opened #1013 as this looks like a bug. Will have a look later and ideally see if I can write a JUnit for it.
